### PR TITLE
Allow testing with basic auth

### PR DESCRIPTION
**Pull request by elvisyjlin**
Support for testing with basic auth

#### Usage
Place the files as
```text
test/[YOUR DEVICE ID].js
test/[YOUR DEVICE ID].cred.json
```

If basic auth is set in the `[YOUR DEVICE ID]/manifest.tt`, you can test the device with specified credentials.

### CODE REVIEW

File: test/index.js [hunk 1]
```diff
@@ -62,6 +62,13 @@ async function loadDeviceFactory(deviceKind) {
 async function createDeviceInstance(deviceKind, factory) {
     if (factory.metadata.auth.type === 'none')
         return new factory(mockEngine, { kind: deviceKind });
+    if (factory.metadata.auth.type === 'basic') {
+        // credentials are stored in test/[DEVICE ID].cred.json
+        const credentialsPath = path.resolve(path.dirname(module.filename), deviceKind + '.cred.json');
+        const args = require(credentialsPath);
```
Comment (gcampax) [inline test/index.js]: if you use `require` you don't need the full path, as `require` resolves the path relative to the calling module by itself.

**Comment by gcampax** [comment]
Also, this solution is ok for manual testing, but won't work in Travis, which does not have the file. It would be good to find a solution for both, maybe based on encrypted files.

File: test/index.js (hunk 1 above)
Comment (elvisyjlin) [reply test/index.js]: Understand. I'll simplify it.

**Comment by elvisyjlin** [comment]
I noticed the error. Does Travis CI test it with basic auth? What is the device?

How about make it support both `[DEVICE ID].cred.json` and `[DEVICE ID].cred.json.enc`? And we can use DEVELOPER_KEY as our encryption key.

**Pull request merged**